graylog2-server/src/main/java/org/graylog/security/certutil/CaServiceImpl.java

```diff
@@ -126,7 +126,7 @@ public void upload(String pass, List<FormDataBodyPart> parts) throws CACreationE
 
     @Override
     public void startOver() {
-
+        keystoreStorage.removeKeyStore(mongoDbCaLocation);
```

Comment:
I'd replace it with:
```
            certificatesService.removeCert(mongoDbCaLocation);

```
And throw away all the remove methods in all the classes.

Reply:
Alternatively, I'd keep the `remove` method in `SmartKeystoreStorage` class only, not making it part of `KeystoreStorage` interface, however.